vowpalwabbit/shared_feature_merger.cc

```diff
@@ -47,7 +47,18 @@ void predict_or_learn(sfm_data& data, VW::LEARNER::multi_learner& base, multi_ex
 
   multi_ex::value_type shared_example = nullptr;
 
-  const bool has_example_header = CB::ec_is_example_header(*ec_seq[0]);
+  bool has_example_header = false;
+  // TODO: Use label type from base when available -- currently hacky method to find label type
+  if (ec_seq[0]->l.cb.costs.size() > 0) { has_example_header = CB::ec_is_example_header(*ec_seq[0]); }
```

Comment:
A reduction should never read more than 1 label is a certain invocation. The label type should be checked in setup and then only that one checked

Reply:
Ok I added a static label_type to the shared_feature_merger namespace which is set in setup

Reply:
It should not be static, that means we can only ever use a single instance of shared_feature_merger

Reply:
Got it, removed static